homeassistant/components/mazda/climate.py

```diff
@@ -0,0 +1,197 @@
+"""Platform for Mazda climate integration."""
+from typing import Any
+
+from pymazda import Client as MazdaAPIClient
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import ClimateEntityFeature, HVACMode
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    PRECISION_HALVES,
+    PRECISION_WHOLE,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.util.temperature import convert as convert_temperature
+
+from . import MazdaEntity
+from .const import DATA_CLIENT, DATA_COORDINATOR, DATA_REGION, DOMAIN
+
+PRESET_DEFROSTER_OFF = "Defroster Off"
+PRESET_DEFROSTER_FRONT = "Front Defroster"
+PRESET_DEFROSTER_REAR = "Rear Defroster"
+PRESET_DEFROSTER_FRONT_AND_REAR = "Front and Rear Defroster"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the switch platform."""
+    client = hass.data[DOMAIN][config_entry.entry_id][DATA_CLIENT]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][DATA_COORDINATOR]
+    region = hass.data[DOMAIN][config_entry.entry_id][DATA_REGION]
```

Comment:
I'd probably unpack this first to reduce the number of dict lookups python has to do.

Ideally we store the data as a `dataclass` instead of `dict` so it has better typing

domain_data = hass.data[DOMAIN][config_entry.entry_id]
client = domain_data[DATA_CLIENT]

Reply:
Thanks, I have updated the code to unpack the dict first.

I agree with your `dataclass` suggestion. I'm planning to open a general cleanup PR after this PR is merged, and I'll make sure to include that change. By the way, do you know of any other integrations that use a `dataclass` in this way, so I can use them as an example?

Reply:
`lookin` has an example

Reply:
Thanks. I'll refer to that when I work on the follow-up PR.